Sources/NIO/CircularBuffer.swift

```diff
@@ -52,6 +53,11 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
         assert(self.buffer.count == capacity)
     }
 
+    /// Allocates an empty buffer.
+    public init() {
+        self.buffer = ContiguousArray<E?>()
+    }
```

Comment:
Do we need the extra `init`?

Reply:
Unfortunately; it is required by `RangeReplaceableCollection`

Reply:
Ah, of course it is. That's unfortunate.

Reply:
@esttorhe we should only have one `init` function that actually sets the member variables. The empty init should maybe pick a 'sensible' initial ring capacity. What about just doing:

    self.init(initialRingCapacity: 16)

or something?

Reply:
Makes sense; let me address this in the next commits @weissi